### PR TITLE
[IMP] mail: clean legacy activity file

### PR DESCRIPTION
**Pull request by seb-odoo**
In preparation of future clean up.

- remove obsolete method _fetchSpecialActivity
- apply lint / some es6 syntax on legacy activity file
- use async/await and remove most self in legacy activity
- inline activity setFileUploadID function
- inline activity _bindOnUploadAction function
- inline activity setDelayLabel function
- inline activity _readActivities function
- inline activity _sendActivityFeedback function
- inline activity _bindPopoverFocusout function
- remove obsolete activity _unbindOnUploadAction function

### CODE REVIEW

File: addons/mail/static/src/js/activity.js [hunk 1]
```diff
@@ -254,6 +233,16 @@ const BasicActivity = AbstractField.extend({
                 $popover.addClass('o_mail_activity_feedback').attr('tabindex', 0);
                 $popover.find('#activity_feedback').focus();
                 self._bindPopoverFocusout($(this));
```
Comment (alexkuhn) [inline addons/mail/static/src/js/activity.js:235]: forgot to remove function

File: addons/mail/static/src/js/activity.js [hunk 2]
```diff
@@ -394,19 +394,6 @@ const BasicActivity = AbstractField.extend({
         });
         this._reload.bind({ activity: true });
     },
-    /**
-     * Unbind event triggered when a file is uploaded.
-     *
-     * @private
-     * @param {Array} activities: list of activity to unbind
-     */
-    _unbindOnUploadAction(activities) {
```
Comment (alexkuhn) [inline addons/mail/static/src/js/activity.js]: ok, it's done in following commit, so ignore

File: addons/mail/static/src/js/activity.js (hunk 1 above)
Comment (seb-odoo) [reply addons/mail/static/src/js/activity.js:235]: Good catch, so that part is not tested, good to know...

**Review by alexkuhn — APPROVED** [review]

**Comment by seb-odoo** [comment]
@robodoo override=ci/security

Code move, and the content is coming from the render of a template which is safe.

**Comment by alexkuhn** [comment]
@robodoo r+

**Pull request closed without merging**